open-ce/validate_config.py

```diff
@@ -46,27 +46,28 @@ def validate_env_config(conda_build_config, env_config_files, variants, reposito
                                                variant['cudatoolkit'],
                                                repository_folder=repository_folder,
                                                conda_build_config=conda_build_config)
-                validate_build_tree(recipes, variant)
+                variant_string = utils.variant_string(variant["python"], variant["build_type"],
+                                               variant["mpi_type"], variant["cudatoolkit"])
+
+                validate_build_tree(recipes, variant_string)
             except OpenCEError as err:
                 raise OpenCEError(Error.VALIDATE_CONFIG, conda_build_config, env_file, variant, err.msg) from err
             print('Successfully validated {} for {} : {}'.format(conda_build_config, env_file, variant))
 
-def validate_build_tree(recipes, variant):
+def validate_build_tree(recipes, variant_string):
     '''
     Check a build tree for dependency compatability.
     '''
-    packages = [package for recipe in recipes for package in recipe.packages]
     channels = {channel for recipe in recipes for channel in recipe.channels}
-    deps = {dep for recipe in recipes for dep in recipe.run_dependencies}
-    deps.update(recipes.get_external_dependencies(variant))
 
-    pkg_args = " ".join(["\"{}\"".format(utils.generalize_version(dep)) for dep in deps
-                                                                    if not utils.remove_version(dep) in packages])
```

Comment:
While working on tests, I found that one of the tests from validate_config_test.py is failing because we don't expect the package from env file to be present in the cli_string while verifying `conda create --dry-run... `. I think the main package itself should always be there while environment creation. Is validate_config meant to verify only dependencies's installation in an env or the main package along with its dependencies?

Reply:
Reverting above change fixes the test.